src/qml/mixxxcontroller.h

```diff
@@ -0,0 +1,30 @@
+#ifndef MIXXX_MIXXXCONTROLLER_H
+#define MIXXX_MIXXXCONTROLLER_H
+
+#include <QtQml/qqmlregistration.h>
+
+#include <QObject>
+
+namespace mixxx {
+namespace qml {
+
+class MixxxController : public QObject {
+    Q_OBJECT
+    QML_ELEMENT
+    Q_PROPERTY(QString controllerId READ controllerId WRITE setControllerId)
+    Q_PROPERTY(bool debugMode READ debugMode WRITE setDebugMode)
+  public:
+    QString controllerId();
```

Comment:
So I get a:

```
/usr/bin/ld: .../moc_mixxxcontroller.cpp:119:(.text+0x177): undefined reference to `mixxx::qml::MixxxController::controllerId()'
```

on this. It's not clear to me whether the `Q_PROPERTY` macro generates the accessors definition or not. If not, I don't really get the usage of this macro…

Reply:
> It's not clear to me whether the Q_PROPERTY macro generates the accessors definition or not. It does not generate accessors. Its there to wire up the classic C++ setter/getter and signal logic to a single property available to QML. This does usually result in a lot of boilerplate. If you don't need the explicit access control, you can just specify a `MEMBER` and eliminate the need for accessors. 

> So I get a:
the issue is that you _declared_ a `controllerId` member function but never _defined_ (implemented) it.

Reply:
see https://doc.qt.io/qt-6/properties.html